network/serde/jsonrpc.go

```diff
@@ -17,60 +31,49 @@ const (
 	TypeError    MessageType = 3
 )
 
-type JsonRpcRequest struct {
-	Jsonrpc string      `json:"jsonrpc"`
-	Id      uint64      `json:"id"`
-	Method  string      `json:"method"`
-	Params  interface{} `json:"params"`
+type JsonRpcRequestResponse struct {
+	Jsonrpc string        `json:"jsonrpc"`
+	Id      uint64        `json:"id"`
+	Method  RequestMethod `json:"method"`
+	Params  interface{}   `json:"params"`
+	Result  interface{}   `json:"result"`
+	Error   interface{}   `json:"error"`
 }
-
-type JsonRpcResponse struct {
-	Jsonrpc string      `json:"jsonrpc"`
-	Id      uint64      `json:"id"`
-	Result  interface{} `json:"result"`
-	Error   interface{} `json:"error"`
+type RequestPayload interface {
+	directfund.ObjectiveRequest | directdefund.ObjectiveRequest | virtualfund.ObjectiveRequest | virtualdefund.ObjectiveRequest
 }
-
-type JsonRpcRequestResponse struct {
+type JsonRpcRequest[T RequestPayload] struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Id      uint64 `json:"id"`
+	Method  string `json:"method"`
+	Params  T      `json:"params"`
+}
+type ResponsePayload interface {
```

Comment:
We're not uniform in our response type. `directdefund` and `virtualdefund` just return an `ObjectiveId` instead of an `ObjectiveResponse`.